engine/crates/gateway-core/src/lib.rs

```diff
@@ -109,23 +110,48 @@ where
                 .instrument(info_span!("execute_stream"))
                 .await
         } else {
-            cache::process_execution_response(
-                ctx.as_ref(),
-                cache::execute_with_cache(
-                    &self.cache,
-                    &self.cache_config,
-                    ctx.as_ref(),
-                    request,
-                    auth,
-                    |request, auth| {
-                        Arc::clone(&self.executor)
-                            .execute(Arc::clone(ctx), auth, request)
-                            .instrument(info_span!("execute"))
-                            .map(move |res| res.map(Arc::new))
-                    },
-                )
-                .await,
-            )
+            if !self.cache_config.global_enabled || !self.cache_config.partial_registry.enable_caching {
+                let result = Arc::clone(&self.executor)
+                    .execute(Arc::clone(ctx), auth, request)
+                    .instrument(info_span!("execute"))
+                    .await?;
+
+                return Response::engine(Arc::new(result));
+            }
+
+            match build_cache_key(&self.cache_config, ctx.as_ref(), &request, &auth) {
+                Ok(cache_key) => {
+                    let execution_fut = Arc::clone(&self.executor)
+                        .execute(Arc::clone(ctx), auth, request)
+                        .instrument(info_span!("execute"))
+                        .map(move |res| res.map(Arc::new));
```

Comment:
Don't think this needs to be `move`?
```suggestion
                        .map(|res| res.map(Arc::new));
```